caleb-lab/server.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+
+const bodyParser = require('./lib/body-parser');
+const http = require('http');
+const url = require('url');
+const queryString = require('querystring');
+const cowsay = require('cowsay');
+const PORT = process.env.PORT || 3000;
+
+const server = module.exports = http.createServer(function(req, res){
+  console.log(req.url);
+  req.url = url.parse(req.url);
+  req.url.query = queryString.parse(req.url.query);
+
+  if(req.method === 'POST'){
+    if(req.url.pathname === '/cowsay'){
+      bodyParser(req, function(err){
+        if(err) console.error(err);
+        let message = cowsay.think({text: req.body.text});
+        console.log(res.body);
+        res.writeHead(200, {'Content-Type': 'text/plain' });
+        res.write(message);
+        res.end(); // this will not be here, will be in asynchronous callback(s);
```

Comment:
**TODO FOR LOST POINTS**
I won't comment on every comment and console.log, but a reminder to please take them out before submission.